Pod/Classes/SEGAmplitudeIntegration.m

```diff
@@ -98,62 +98,46 @@ - (void)trackRevenue:(NSDictionary *)properties andRevenueOrTotal:(NSNumber *)re
         [self trackLogRevenueV2:properties andRevenueOrTotal:revenueOrTotal];
-    } else {
-        // fallback to logRevenue v1
-        id productId = [properties objectForKey:@"productId"] ?: [properties objectForKey:@"product_id"];
-        if (!productId || ![productId isKindOfClass:[NSString class]]) {
-            productId = nil;
-        }
-        id quantity = [properties objectForKey:@"quantity"];
-        if (!quantity || ![quantity isKindOfClass:[NSNumber class]]) {
-            quantity = [NSNumber numberWithInt:1];
-        }
-        id receipt = [properties objectForKey:@"receipt"];
-        if (!receipt || ![receipt isKindOfClass:[NSString class]]) {
-            receipt = nil;
-        }
-        [self.amplitude logRevenue:productId
-                          quantity:[quantity integerValue]
-                             price:revenueOrTotal
-                           receipt:receipt];
-        SEGLog(@"[Amplitude logRevenue:%@ quantity:%d price:%@ receipt:%@];", productId, [quantity integerValue], revenueOrTotal, receipt);
+        return;
     }
+
+    // fallback to logRevenue v1
+    NSString *productId = [properties objectForKey:@"productId"] ?: [properties objectForKey:@"product_id"] ?: nil;
+    NSNumber *quantity = [properties objectForKey:@"quantity"] ?: [NSNumber numberWithInt:1];
+    id receipt = [properties objectForKey:@"receipt"] ?: nil;
+    [self.amplitude logRevenue:productId
+                      quantity:[quantity integerValue]
+                         price:revenueOrTotal
+                       receipt:receipt];
+    SEGLog(@"[Amplitude logRevenue:%@ quantity:%d price:%@ receipt:%@];", productId, [quantity integerValue], revenueOrTotal, receipt);
 }
 
 - (void)trackLogRevenueV2:(NSDictionary *)properties andRevenueOrTotal:(NSNumber *)revenueOrTotal
 {
-    id price = [properties objectForKey:@"price"];
-    id quantity = [properties objectForKey:@"quantity"];
-
-    // if no price fallback to using revenue
-    if (!price || ![price isKindOfClass:[NSNumber class]]) {
-        price = revenueOrTotal;
-        quantity = [NSNumber numberWithInt:1];
-    } else if (!quantity || ![quantity isKindOfClass:[NSNumber class]]) {
-        quantity = [NSNumber numberWithInt:1];
-    }
-
+    NSNumber *price = [properties objectForKey:@"price"] ?: revenueOrTotal;
+    NSNumber *quantity = [properties objectForKey:@"quantity"] ?: [NSNumber numberWithInt:1];
     [[self.amprevenue setPrice:price] setQuantity:[quantity integerValue]];
     SEGLog(@"[[AMPRevenue revenue] setPrice:%@] setQuantity: %d];", price, [quantity integerValue]);
 
-    id productId = [properties objectForKey:@"productId"] ?: [properties objectForKey:@"product_id"];
-    if (productId && [productId isKindOfClass:[NSString class]] && ![productId isEqualToString:@""]) {
+    NSString *productId = [properties objectForKey:@"productId"] ?: [properties objectForKey:@"product_id"];
+    if (productId && ![productId isEqualToString:@""]) {
```

Comment:
This is gonna crash if `productId` is not a `NSString` for some reason. You don't have compile time guarantee, so you're better off protecting against it.